### PR TITLE
Enable modernize-use-equals-default and modernize-use-equals-delete i…

### PR DESCRIPTION
**Pull request by steven-johnson**
…n clang-tidy

### CODE REVIEW

File: src/autoschedulers/adams2019/FunctionDAG.h [hunk 1]
```diff
@@ -567,14 +567,15 @@ struct FunctionDAG {
...
+    FunctionDAG(const FunctionDAG &) = delete;
+    FunctionDAG &operator=(const FunctionDAG &) = delete;
+    FunctionDAG(FunctionDAG &&) = delete;
+    FunctionDAG &operator=(FunctionDAG &&) = delete;
+
 private:
     // Compute the featurization for the entire DAG
     void featurize();
 
-    // This class uses a lot of internal pointers, so we'll hide the copy constructor.
```
Comment (abadams) [inline src/autoschedulers/adams2019/FunctionDAG.h:574]: Comment was lost

**Review by abadams — APPROVED** [review]
lgtm pending tests (I'm a little wary of the runtime changes)

**Comment by abadams** [comment]
Build failures due to newly detected unused globals in JITModule.cpp. I could have sworn we already deleted those...

**Comment by steven-johnson** [comment]
> Build failures due to newly detected unused globals in JITModule.cpp. I could have sworn we already deleted those...

https://github.com/halide/Halide/pull/5379 has to land before this one anyway, so let's wait and see how that falls out

**Comment by steven-johnson** [comment]
I'm going to drop and re-create this PR, the merge conflicts are insane

**Pull request closed without merging**